frontend/src/metabase/query_builder/components/filters/FilterPopover/FilterPopover.tsx

```diff
@@ -242,83 +246,103 @@ export function FilterPopover({
   const supportsExpressions = query.database()?.supportsExpressions();
 
   const filterOperator = filter.operator();
-  const hasPicker = filterOperator && filterOperator.fields.length > 0;
+  const hasPicker =
+    (filterOperator && filterOperator.fields.length > 0) || isBoolean(field);
```

Comment:
this caused `empty` and `not-empty` boolean filters to make the picker disappear 🤦

Reply:
I think it might be worth adding a comment explaining the thing
It'd be easy to forget why we had to treat boolean fields differently